### ISSUE TITLE
Custom logic for tests.

### ISSUE THREAD

**Issue by danieletulone** [rangav/thunder-client-support#194]
**Is your feature request related to a problem? Please describe.**

I create a request. I want to make a test that asserts response fields using custom logic.
How to make advanced tests?

**Describe the solution you'd like**

1. Create a request with a name.
2. Define url, headers, body, etc.
3. Define tests
4. Create a file (workspace_path, thunder_path, collection_path + request_name + .js/.ts).
5. Define one or more methods and export them. They will be return true or false, if passed or not.
6. Send request. (thunder-client will run also each test methods defined into js/ts file)

Example:
1. Create a collection named 'My Collection'
2. Create a request named 'My Request' and add to 'My Collection'
3. Create a file for custom test logics ($WORKSPACE/thunder/my-collection/my-request.js)
4. Send request.

my-request.js
```javascript
import { isEmail } from '../my-utils.js'

export function testEmailField(payload) {
    return isEmail(payload.response.body.user.email)
}
```

**Comment by rangav** [comment]
Hi @danieletulone thanks for the suggestion. Will review about how to implement advanced tests.

Will regex match work for you? #57

**Comment by danieletulone** [comment]
Yeah, regex will works but not for all tests. Providing a generic way for include custom code will be a universal solution for advanced tests.

**Comment by rangav** [comment]
Thanks @danieletulone, Will add this to roadmap

**Comment by rangav** [comment]
Regex match feature is now implemented in v1.9.0
https://github.com/rangav/thunder-client-support/releases/tag/v1.9.0

please let me know if any feedback

**Comment by joaonunogomes** [comment]
+1 for this option.
Is this planned for a future release ?
What I'm looking for is a generic way to do more advanced tests, like referring to a custom script or even writing the script on the test request tab.  

Great work so far, thanks!

**Comment by rangav** [comment]
This feature is now implemented and published to marketplace, please update to v2.0.0

See all features released
https://github.com/rangav/thunder-client-support/releases/tag/v2.0.0

Please let me know your feedback.

**Comment by rangav** [comment]
Hi, Now Tests via scripting is possible with v2.10.0

https://github.com/rangav/thunder-client-support/releases/tag/v2.10.0
